developer-docs/adding-gem-dependencies.md

```diff
@@ -0,0 +1,32 @@
+# Adding gem dependencies
+
+When new gem dependencies are added to Bolt, it's important to add them to both the bolt-runtime and pe-bolt-server-runtime. You can add the gem dependencies by following these steps:
+
+1. Open a PR in the [puppet-runtime repo](https://github.com/puppetlabs/puppet-runtime) with the following changes:
+
+    * Add each gem to the [config components directory](https://github.com/puppetlabs/puppet-runtime/tree/master/configs/components). The file for a gem component should be named `rubygem-<GEM_NAME>.rb`.
+
+        ```ruby
+        component "rubygem-<GEM_NAME>" do |pkg, settings, platform|
+          pkg.version "<GEM_VERSION>"
+          pkg.md5sum "<GEM_MD5SUM>"
+
+          instance_eval File.read('configs/components/_base-rubygem.rb')
+        end
+        ```
+
+    * Add the components to the [bolt-runtime](https://github.com/puppetlabs/puppet-runtime/blob/master/configs/projects/bolt-runtime.rb) and [_shared-pe-bolt-server](https://github.com/puppetlabs/puppet-runtime/blob/master/configs/projects/_shared-pe-bolt-server.rb).
```

Comment:
Also need to add it to artifactory mirror. We do that with kerminator in `#release-new-new` 
```
!mirrorsource https://rubygems.org/downloads/yard-0.9.20.gem
```